synapse/util/caches/__init__.py

```diff
@@ -204,15 +206,15 @@ def intern_string(string):
         return string
 
 
-def intern_dict(dictionary):
+def intern_dict(dictionary: Dict[str, Any]) -> Dict[str, Any]:
     """Takes a dictionary and interns well known keys and their values"""
     return {
         KNOWN_KEYS.get(key, key): _intern_known_values(key, value)
         for key, value in dictionary.items()
     }
 
 
-def _intern_known_values(key, value):
+def _intern_known_values(key: str, value: Any) -> Any:
```

Comment:
should this be `value: Optional[str]`? Or does that only apply for the when `key in intern_keys`?

Reply:
I don't think so, this is used by `intern_dict` which is called over event JSON blobs, so the values can be things besides strings or null.

We do seem to be assuming that the *values* for `intern_keys` will be strings though, but I didn't want to mess with the logic in here.